lib/src/runner/engine.dart

```diff
@@ -529,8 +529,8 @@ class Engine {
   Future close() async {
     _closed = true;
     if (_closedBeforeDone != null) _closedBeforeDone = true;
-    _onSuiteAddedController.close();
-    _suiteController.close();
+    await _onSuiteAddedController.close();
+    await _suiteController.close();
```

Comment:
I removed this await and did a couple Travis runs. Unfortunately I could not get into the flaky state.